pkg/controller/pcideviceclaim/pcideviceclaim_controller.go

```diff
@@ -107,6 +108,8 @@ func (h *Handler) OnRemove(name string, pdc *v1beta1.PCIDeviceClaim) (*v1beta1.P
 		return pdc, err
 	}
 
+	lock.Lock()
```

Comment:
any reasons why the lock is requried on the remove method?

Reply:
The controller is running two threads by default.

The locking was introduced to ensure that device plugin updates are serialized.

The remove is also managed by a lock, as the plugin is deleted if there are no more pcideviceclaims for the device type and we need to ensure that plugin is working until the last deviceclaim is deleted.

Reply:
Adding this as a code comment will be helpful, thanks.

Reply:
done.